src/datasources/event.ts

```diff
@@ -1,19 +1,27 @@
+import { LinkedEventsSource } from '../types/types';
 import { basePaths } from './constants';
 import DataSource from './LinkedEventsDataSource';
 
 class EventAPI extends DataSource {
```

Comment:
For class it would be more natural to use constructor that sets the defaultSource, then you dont have to pass it to every method call. Or change this to a function

Reply:
I don't understand what you mean. What are we passing in every method call now?

Reply:
It's just something that comes into mind if I see this class-file and dont know how is it actually used:

```
const api = EventApi(source1);
api.get('/path');
api. getEventDetails(id,query);
...
const api2 = EventApi(source2);
api2.get('/path');
api2. getEventDetails(id,query);
```
Would feel like more natural than:
```
const api = EventApi();
api.get('/path',source1);
api.getEventDetails(id,query,source1);
api.get('/path',source2);
api.getEventDetails(id,query,source2);
```
It's not a big thing and no need to change :)

Reply:
The thing is that source is optional parameter that we can use to override the default one :)

Reply:
And in this context it is only initialized once for the apollo context

Reply:
Let's keep it this way